### ISSUE TITLE
Handle evidence file errors better

### ISSUE THREAD

**Issue by afaulconbridge** [opentargets/platform#378]
In some cases, there may be an IO error accessing an evidence source (e.g. 404 from a url, non existent file, transport error, etc). Currently this is not handled gracefully. Instead what should happen:

Immediately log a warning.
Store the filename/url of the error for later
Continue processing the remaining file
At the end of the process, print error level messages for each failed source and then exit with **non-zero exit code**

**Comment by Docproc** [comment]
An evidence file which is missing, empty, or corrupt can also cause this; the symptoms are that the validation seems to hang indefinitely, with little or no processor or memory usage.

Note that an empty file when gzipped has a non-zero size (20 bytes or so) and will also trigger this issue.

**Comment by Docproc** [comment]
The only error that is shown in these situations (which is often obscured by later log messages) is
```
2018-12-23 14:05:05,750 - mrtarget.common.IO - DEBUG - generate an iterator of (filename,enumerate) for filename /Users/glennproctor/work/opentargets_1/data_pipeline/json/missing.json.gz
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/Users/glennproctor/.virtualenvs/data_pipeline/lib/python2.7/site-packages/pypeln/process.py", line 706, in _from_iterable
    _run_task(f_task, params)
  File "/Users/glennproctor/.virtualenvs/data_pipeline/lib/python2.7/site-packages/pypeln/process.py", line 307, in _run_task
    f_task(args)
  File "/Users/glennproctor/.virtualenvs/data_pipeline/lib/python2.7/site-packages/pypeln/process.py", line 703, in f_task
    for x in iterable:
  File "/Users/glennproctor/.virtualenvs/data_pipeline/lib/python2.7/site-packages/more_itertools/more.py", line 474, in with_iter
    with context_manager as iterable:
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/contextlib.py", line 17, in __enter__
    return self.gen.next()
  File "mrtarget/common/__init__.py", line 103, in open
    f.raise_for_status()
  File "/Users/glennproctor/.virtualenvs/data_pipeline/lib/python2.7/site-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
HTTPError: 404 Client Error: None for url: None
```
which neither states that the error was due to a missing file, nor gives the name of the file.

**Comment by afaulconbridge** [comment]
opentargets/data_pipeline#412 assumes that all evidence files are URLs, and that if it can be initially fetched then there will be no subsequent problems - this might not be the case. Therefore, further work and a PR will be needed to fix the issues introduced.

**Comment by afaulconbridge** [comment]
Fixes as part of opentargets/data_pipeline#420 done. It still checks the URLs at the start of the process, which doesn't guarantee that the process will complete successfully, but its better than before. If we start to see issues in future, we can open a new issue to investigate then.
